src/sql/parts/dashboard/common/dashboardPage.component.ts

```diff
@@ -138,20 +138,29 @@ export abstract class DashboardPage extends Disposable implements OnDestroy {
 
 		let allTabs = dashboardHelper.filterConfigs(dashboardRegistry.tabs, this.dashboardService);
 
-		// Load always show tabs
-		let alwaysShowTabs = allTabs.filter(tab => tab.alwaysShow);
-		this.loadNewTabs(alwaysShowTabs);
+		// Load tab setting configs
+		this._tabSettingConfigs = this.dashboardService.getSettings<Array<TabSettingConfig>>([this.context, 'tabs'].join('.'));
 
-		// Load pinned tabs
-		this._pinnedTabs = this.dashboardService.getSettings<Array<PinConfig>>([this.context, 'tabs'].join('.'));
 		let pinnedDashboardTabs: IDashboardTab[] = [];
-		this._pinnedTabs.forEach(pinnedTab => {
-			let tab = allTabs.find(i => i.id === pinnedTab.tabId);
-			if (tab) {
-				pinnedDashboardTabs.push(tab);
+		let alwaysShowTabs = allTabs.filter(tab => tab.alwaysShow);
+
+		this._tabSettingConfigs.forEach(config => {
+			if (config.tabId && types.isBoolean(config.isPinned)) {
+				let tab = allTabs.find(i => i.id === config.tabId);
+				if (tab) {
+					if (config.isPinned) {
+						pinnedDashboardTabs.push(tab);
+					} else {
+						// overwrite always show if specify in user settings
+						let index = alwaysShowTabs.findIndex(i => i.id === tab.id);
+						alwaysShowTabs.splice(index, 1);
```

Comment:
@karlb@microsoft.com: The tab setting config will overwrite always show. It means that if it has been unpinned, it will be removed from "alwaysShowTabs"